cyclonedx-lib/src/temurin/sbom/TemurinGenSBOM.java

```diff
@@ -178,16 +180,20 @@ static Bom addComponent(final String compName, final String name, final String v
         bom.addComponent(comp);
         return bom;
     }
-    static Bom addComponentProperty(final String name, final String value, final String fileName) {     // Method to store Component --> Property --> name-value pairs
+    static Bom addComponentProperty(final String compName, final String name, final String value, final String fileName) {     // Method to add Component --> Property --> name-value pairs
         Bom bom = readJSONfile(fileName);
-        Property prop1 = new Property();
-        List<Property> prop = new ArrayList<>();
-        Component comp = new Component();
-        prop1.setName(name);
-        prop1.setValue(value);
-        prop.add(prop1);
-        comp.setProperties(prop);
-        bom.addProperty(prop1);
+        List<Component> componentArrayList = bom.getComponents();
+        for (Component item : componentArrayList) {
+            if (Objects.equals(item.getName(), compName)) {
+                    List<Property> propertiesList = item.getProperties();
+                    ArrayList<Property> newPropertyList = new ArrayList<>(propertiesList);
+                    Property prop1 = new Property();
+                    prop1.setName(name);
+                    prop1.setValue(value);
+                    newPropertyList.add(prop1);
+                    item.setProperties(newPropertyList);
```

Comment:
Instead of getting the propertyList then adding to it, simply use the component.addProperty(Property) method, as you will actually be replacing the existing property list in the Component rather than just adding to it.
ie:
```
item.addProperty(prop1);
```

Reply:
@andrew-m-leonard we have to make a list specifically to add more name-value pairs. I tried using `item.addProperty(prop1)` before. It doesn't add to the component property list.

Reply:
@SehrishHussain the issue with addProperty() has been fixed in cyclonedx v7.1.0. So lets move up to that latest release..